src/components/homePage/HomePage.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useRef, useState } from 'react'
+
+import InfiniteScroll from 'react-infinite-scroll-component'
+
+import './HomePage.css'
+
+import BookItem from '../../interfaces/BookItem'
+import { getBooksPaged } from '../../services/book.service'
+import BookList from '../bookList/BookList'
+
+const initialPageLenght  = 12
+
+interface Props {
+  searchInput : string
+}
+
+function HomePage( { searchInput } : Props) {
+  const [ bookList, setBookList ] = useState<BookItem[]>([])
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ hasMore, setHasMore ] = useState(false)
+  const currentSearch = useRef<string>(searchInput)
+
+  const addNextPage = () => {
+    setPageNumber(pageNumber + 1)
+  }
+
+  const feachBooks = () => {
+    getBooksPaged({ PageNumber: pageNumber, PageLength: initialPageLenght, Where: [ { Field: 'Title', Value: searchInput, Operation: 2 } ] }).then((response =>
```

Comment:
You should have pageable request type instead of just using this object without type